### PR TITLE
Feature/audit log fetching

### PR DESCRIPTION
**Pull request by TopiSenpai**

### CODE REVIEW

**Comment by Skye-31** [comment]
is this ready for review or should I put it in draft?

**Comment by TopiSenpai** [comment]
should be ready to review 
was about to do one myself first

File: api/webhook.go [hunk 1]
```diff
@@ -12,17 +12,17 @@ const (
...
+	Username      Snowflake     `json:"username"`
+	GuildID       *Snowflake    `json:"guild_id"`
+	ChannelID     Snowflake     `json:"channel_id"`
+	User          *User         `json:"user"`
+	Name          string        `json:"name"`
+	Avatar        string        `json:"avatar"`
+	Token         *string       `json:"token"`
+	ApplicationID *Snowflake    `json:"application_id"`
+	SourceGuild   *PartialGuild `json:"source_guild"`
+	SourceChannel *TextChannel  `json:"source_channel"`
```
Comment (TopiSenpai) [inline api/webhook.go:26]: this should be a partial channel

**Comment by Skye-31** [comment]
has conflicts now anyway

**Review by TopiSenpai — COMMENTED** [review]
maybe some async helper?

File: api/webhook.go (hunk 1 above)
Comment (TopiSenpai) [reply api/webhook.go]: alright done

File: api/guild.go [hunk 2]
```diff
@@ -259,6 +259,11 @@ func (g *Guild) IconURL(size int) *string {
 	return &u
 }
 
+// GetAuditLogs gets AuditLog(s) for this Guild
+func (g *Guild) GetAuditLogs(userID Snowflake, actionType AuditLogEvent, before Snowflake, limit int) (*AuditLog, restclient.RestError) {
```
Comment (Skye-31) [inline api/guild.go:263]: Shouldn't this return more than one?

Comment (TopiSenpai) [reply api/guild.go:263]: nope
that is just a wrapper struct as described in the docs <https://discord.com/developers/docs/resources/audit-log#get-guild-audit-log>

**Review by Skye-31 — APPROVED** [review]
LGTM

**Pull request merged**